book/src/assets/how_to_define_custom_formats.md

```diff
@@ -40,20 +48,17 @@ If you are defining a new format that may be useful to others, [please send us a
     #[derive(Clone, Copy, Debug, Default)]
     pub struct MyLangFormat;
 
-    impl<A> SimpleFormat<A> for MyLangFormat
+    impl<D> Format<D> for MyLangFormat
```

Comment:
In the light of changes that happened to formats, i'm not sure if this is really the thing we should be explaning first. Generic formats are kinda tricky, as those are basically a proxy for serialization format like ron or json. It's kinda a hack really. The second, i would call "more idiomatic" group of formats are actually solving the problem "i have some data and i want to transform it to soemthing that can be loaded". Those formats specify their "output type" directly, like ObjFormat that is converting a blob with obj data into MeshBuilder. This is what i would consider you want to add more often, like FbxFormat or BmpFormat could be a thing.

But then, do we even care at this point? This will likely change again once atelier is in place, so maybe we defer that.

Reply:
For now, I just want to fix the book to be correct. I think I agree this can be improved, but I'll just leave it like this for now since the scope of this PR is only to get the book correct.